python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -0,0 +1,276 @@
+from datetime import datetime
+from typing import TYPE_CHECKING, AbstractSet, Mapping, NamedTuple, Optional
+
+from typing_extensions import TypeAlias
+
+from dagster import _check as check
+from dagster._core.definitions.asset_subset import AssetSubset, ValidAssetSubset
+from dagster._core.definitions.events import AssetKey, AssetKeyPartitionKey
+from dagster._utils.cached_method import cached_method
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.base_asset_graph import BaseAssetGraph
+    from dagster._core.definitions.data_version import CachingStaleStatusResolver
+    from dagster._core.definitions.definitions_class import Definitions
+    from dagster._core.definitions.partition import PartitionsDefinition
+    from dagster._core.instance import DagsterInstance
+    from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
+
+
+class TemporalContext(NamedTuple):
+    """TemporalContext represents an effective time, used for business logic, and last_event_id
+    which is used to identify that state of the event log at some point in time. Put another way,
+    the value of a TemporalContext represents a point in time and a snapshot of the event log.
+
+    Effective time: This is the effective time of the computation in terms of business logic,
+    and it impacts the behavior of partioning and partition mapping. For example, if an
+    if you get the "last" partition window of a given partitions definition, it is with
+    respect to the effective time.
+
+    Last event id: Our event log has a monotonic increasing event id. This is used to
+    cursor the event log. This event_id is also propogated to derived tables to indicate.
+    This allows us to query the state of the event log at a given point in time.
+
+    Note that insertion time of the last_event_id is not the same as the effective time.
+
+    A last_event_id of None indicates that the reads will be volatile will immediately
+    reflect any subsequent writes.
+    """
+
+    effective_dt: datetime
+    last_event_id: Optional[int]
```

Comment:
Naming opinion: Since `TemporalContext` represents a point in time/event history, IMO these should just be `datetime` and `event_id`-- the qualifiers "effective" and "last" are already implied by the fact that these are on a `TemporalContext`.

Reply:
Yeah the only issue here is that there values will appear on their own without a TemporalContext object in lots of places.

Reply:
Yeah I understand e.g. "effective datetime" will became a new standard phrase, but I think my reasoning still holds. Like if I had a cursor class that represented an x, y upper bound:

```
cursor = MaxCursor(x=..., y=...)  # not max_x, max_y

# then later if I need to extract one of these values

max_x = cursor.x
```

I think the situation is similar here. Just kinda a matter of taste though, NBD.